src/appwindow.cpp

```diff
@@ -158,12 +158,34 @@ void AppWindow::finishConstruction()
         setWindowOpacity(1);
 #endif
 
-    if (SettingsHelper::isFirstTimeUser())
-    {
-        LOG_INFO("Is first-time user");
-        preferencesWindow->display();
-        SettingsHelper::setFirstTimeUser(false);
-    }
+    // The window needs time to make its geometry stable. We wait it to display the new dialogs in correct positions
+    QTimer::singleShot(200, [this] {
+        if (SettingsHelper::isFirstTimeUser())
+        {
+            LOG_INFO("Is first-time user");
+            preferencesWindow->display();
+            SettingsHelper::setFirstTimeUser(false);
+        }
+        else if (!SettingsHelper::isPromotionDialogShown() && SettingsHelper::getTotalUsageTime() >= 36000)
```

Comment:
```suggestion
        else if (!SettingsHelper::isPromotionDialogShown() && SettingsHelper::getTotalUsageTime() >= 10 * 60 * 60) // 10 hours or above
```

This makes it easy to change and read.